cockatrice/src/spoilerbackgroundupdater.cpp

```diff
@@ -0,0 +1,241 @@
+
+    // Force the calling of actSeeIfSpoilerSeasonIsActive() in sequence instead of parallel
+    // as we need to get that result before we start this timer thread up.
+    connect(lpNetworkReply, SIGNAL(finished()), this, SLOT(actSeeIfSpoilerSeasonIsActive()));
+}
+
+void SpoilerBackgroundUpdater::actDownloadFinishedSpoilersFile()
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QString lsSpoilerSavePath = settingsCache->getSpoilerDatabasePath();
+    
+    if (leErrorCode != QNetworkReply::NoError)
+    {
+        QMessageBox::critical(nullptr, tr("Error"), tr("Network error: %1.").arg(lpNetworkReply->errorString()));
+        lpNetworkReply->deleteLater();
+        return;
+    }
+
+    int lnStatusCode = lpNetworkReply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+    if (lnStatusCode == 301 || lnStatusCode == 302)
+    {
+        // Was a redirect, will need to start over!
+        QUrl lcRedirectUrl = lpNetworkReply->attribute(QNetworkRequest::RedirectionTargetAttribute).toUrl();
+        downloadSpoilersFile(lcRedirectUrl);
+    }
+    else
+    {
+        // Read downloaded file into buffer
+        mcDownloadedFileData = lpNetworkReply->readAll();
+
+        if (saveSpoilersToFile(lsSpoilerSavePath))
+        {
+            // Reload the card database to import the spoilers
+            QtConcurrent::run(db, &CardDatabase::loadCardDatabases);
+            QMessageBox::information(nullptr, tr("Success"), tr("Spoiler database updated and saved successfully to\n%1").arg(lsSpoilerSavePath));
+        }
+        else
+        {
+            QMessageBox::critical(nullptr, tr("Error"), tr("Spoiler database updated but could not be saved to %1").arg(lsSpoilerSavePath));
+        }
+    }
+
+    lpNetworkReply->deleteLater();
+}
+
+void SpoilerBackgroundUpdater::actSeeIfSpoilerSeasonIsActive()
+{
+    QNetworkReply *lpNetworkReply = static_cast<QNetworkReply *>(sender());
+    QNetworkReply::NetworkError leErrorCode = lpNetworkReply->error();
+    QString lsSpoilerSavePath = settingsCache->getSpoilerDatabasePath();
+
+    if (leErrorCode != QNetworkReply::NoError)
+    {
+        // If there's a network issue
+        mbIsActiveSpoilerSeason = false;
+        QMessageBox::critical(nullptr, tr("Error"), tr("Network error: %1.").arg(lpNetworkReply->errorString()));
+    }
+    else
+    {
+        // If the file reads "enabled" then we're in spoiler season (true)
+        // If the file reads anything else then we're NOT in spoiler season (false)
+        QByteArray laFileData = lpNetworkReply->readAll();
+        mbIsActiveSpoilerSeason = QString(laFileData).contains("enabled", Qt::CaseInsensitive);
+
+        if (mbIsActiveSpoilerSeason)
+        {
+            // Start the timer for spoilers downloads
+            mpTimerForSpoilers = new QTimer(this);
```

Comment:
runTimer() already takes care of creating the timer if it doesn't exists yet, no need to do it here
(Also, this was not checking if the timer already existed, so it could create a duplicate)